python/config_settings/config_settings.bzl

```diff
@@ -15,26 +15,52 @@
 """This module is used to construct the config settings in the BUILD file in this same package.
 """
 
+load("@bazel_skylib//lib:selects.bzl", "selects")
 load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
 
 # buildifier: disable=unnamed-macro
-def construct_config_settings(python_versions):
+def construct_config_settings(minor_mapping, python_versions):
     """Constructs a set of configs for all Python versions.
 
     Args:
-        python_versions: The Python versions supported by rules_python.
+        minor_mapping: mapping from minor (x.y) versions to the corresponding full (x.y.z) versions.
+        python_versions: list of full (x.y.z) Python versions supported by rules_python.
     """
+
+    minor_versions = list(minor_mapping.keys())
+    allowed_flag_values = python_versions + minor_versions
+
     string_flag(
         name = "python_version",
         build_setting_default = python_versions[0],
```

Comment:
Using `build_setting_default = python_versions[0]` here looks like a minor bug, BTW.

It enables a config setting, and if user manages to configure a toolchain that matches this full version, this toolchain will be selected by default, instead of the one marked by the `is_default` attr in `python.toolchain(...)`.

I think a better approach would be to have something like `build_setting_default = "default"` and `values = ["default"] + allowed_flag_values`.

Reply:
Added a comment to note this